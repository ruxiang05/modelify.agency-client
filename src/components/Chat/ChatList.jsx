import React from 'react';
import ChatCard from './ChatCard';
import PageHeader from '../PageHeader';
import api from '../../api';
import { getToken } from '../../auth';
import '../../styles/chat-list.scss';
import NoData from '../NoData';

class ChatList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      chats: [],
    };
  }

  componentDidMount() {
    const token = getToken();
    api.chats.getChats(token).then((res) => {
      const { chats } = res;
      console.log(res);
      this.setState({ chats });
    });
  }

  render() {
    const { chats } = this.state;
    return (
      <React.Fragment>
        <PageHeader title="Chats" />
        {chats.length ? (
          <ul className="page chats">
            {chats.map(chat => (
              <li key={chats.indexOf(chat)}>
                <ChatCard chat={chat} />
              </li>
            ))}
          </ul>
        ) : (
          <NoData>
            <p>You do not have chats yet</p>
          </NoData>
        )}
      </React.Fragment>
    );
  }
}

export default ChatList;
