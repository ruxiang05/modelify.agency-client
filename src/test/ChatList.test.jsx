import React from 'react';
import { shallow } from 'enzyme';
import ChatList from '../components/Chat/ChatList';

jest.mock('../api.js');

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ChatList />);
});

describe('ChatList', () => {
  it('should render ChatList', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
