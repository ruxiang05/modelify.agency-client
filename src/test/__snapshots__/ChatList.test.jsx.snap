// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList should render ChatList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader
          backButton={false}
          title="Chats"
        />,
        <ul
          className="page chats"
        >
          <li>
            <ChatCard
              chat={
                Object {
                  "id": "2t3457345",
                  "job": Object {},
                  "messages": Array [],
                  "users": Array [
                    "1",
                    "2",
                  ],
                }
              }
            />
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": undefined,
          "actionAltText": undefined,
          "actionIcon": undefined,
          "backButton": false,
          "goBackAction": undefined,
          "history": undefined,
          "title": "Chats",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <ChatCard
                chat={
                  Object {
                    "id": "2t3457345",
                    "job": Object {},
                    "messages": Array [],
                    "users": Array [
                      "1",
                      "2",
                    ],
                  }
                }
              />
            </li>,
          ],
          "className": "page chats",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <ChatCard
                chat={
                  Object {
                    "id": "2t3457345",
                    "job": Object {},
                    "messages": Array [],
                    "users": Array [
                      "1",
                      "2",
                    ],
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "chat": Object {
                  "id": "2t3457345",
                  "job": Object {},
                  "messages": Array [],
                  "users": Array [
                    "1",
                    "2",
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={false}
            title="Chats"
          />,
          <ul
            className="page chats"
          >
            <li>
              <ChatCard
                chat={
                  Object {
                    "id": "2t3457345",
                    "job": Object {},
                    "messages": Array [],
                    "users": Array [
                      "1",
                      "2",
                    ],
                  }
                }
              />
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": false,
            "goBackAction": undefined,
            "history": undefined,
            "title": "Chats",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <ChatCard
                  chat={
                    Object {
                      "id": "2t3457345",
                      "job": Object {},
                      "messages": Array [],
                      "users": Array [
                        "1",
                        "2",
                      ],
                    }
                  }
                />
              </li>,
            ],
            "className": "page chats",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <ChatCard
                  chat={
                    Object {
                      "id": "2t3457345",
                      "job": Object {},
                      "messages": Array [],
                      "users": Array [
                        "1",
                        "2",
                      ],
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "chat": Object {
                    "id": "2t3457345",
                    "job": Object {},
                    "messages": Array [],
                    "users": Array [
                      "1",
                      "2",
                    ],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
